cylc/flow/scripts/cylc.py

```diff
@@ -16,10 +16,36 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 """cylc main entry point"""
 
-import argparse
-from contextlib import contextmanager
 import os
 import sys
+
+
+def pythonpath_manip():
+    """Stop PYTHONPATH contaminating the Cylc Environment
+
+    * Remove PYTHONPATH items from sys.path to prevent PYTHONPATH
+      contaminating the Cylc Environment.
+    * Re-add items from CYLC_PYTHONPATH to sys.path.
+
+    See Also:
+        https://github.com/cylc/cylc-flow/issues/5124
+    """
+    if 'CYLC_PYTHONPATH' in os.environ:
+        for item in os.environ['CYLC_PYTHONPATH'].split(os.pathsep):
+            abspath = os.path.abspath(item)
+            sys.path.insert(0, abspath)
+    if 'PYTHONPATH' in os.environ:
+        os.environ['CYLC_PYTHONPATH'] = os.environ['PYTHONPATH']
```

Comment:
```suggestion
        if 'CYLC_PYTHONPATH' in os.environ:
            os.environ['CYLC_PYTHONPATH'] = ':'.join([
                os.environ['PYTHONPATH'], os.environ['CYLC_PYTHONPATH']
            ])
        else:
            os.environ['CYLC_PYTHONPATH'] = os.environ['PYTHONPATH']
```

Reply:
@MetRonnie  - Does this make sense?

Reply:
Why is this needed at all? Are we not ignoring `PYTHONPATH` now?

Reply:
Yes, I think that the original elif is fine.

@hjoliver  - see this>

Reply:
Not sure if I got my wires crossed this morning, or if the code has changed so I can't see what I thought I was querying anymore.  Sorry if the former!